samples/src/main/java/TableStatus.java

```diff
@@ -19,23 +19,23 @@
 public class TableStatus {
     public static void main(String[] args) {
         try {
-
             Integer timeoutInSec = Integer.getInteger("timeoutInSec");
 
             ConnectionStringBuilder csb =
                     ConnectionStringBuilder.createWithAadApplicationCredentials(System.getProperty("clusterPath"),
                             System.getProperty("appId"),
                             System.getProperty("appKey"),
                             System.getProperty("appTenant"));
-            IngestClient client = IngestClientFactory.createClient(csb);
-
-            IngestionProperties ingestionProperties = new IngestionProperties(System.getProperty("dbName"),
-                    System.getProperty("tableName"));
-            ingestionProperties.setIngestionMapping(System.getProperty("dataMappingName"), IngestionMapping.IngestionMappingKind.Json);
-            ingestionProperties.setReportMethod(QueueAndTable);
-            ingestionProperties.setReportLevel(IngestionProperties.IngestionReportLevel.FailuresAndSuccesses);
-            FileSourceInfo fileSourceInfo = new FileSourceInfo(System.getProperty("filePath"), 0);
-            IngestionResult ingestionResult = client.ingestFromFile(fileSourceInfo, ingestionProperties);
+            IngestionResult ingestionResult;
+            try (IngestClient client = IngestClientFactory.createClient(csb)) {
+                IngestionProperties ingestionProperties = new IngestionProperties(System.getProperty("dbName"),
+                        System.getProperty("tableName"));
+                ingestionProperties.setIngestionMapping(System.getProperty("dataMappingName"), IngestionMapping.IngestionMappingKind.Json);
+                ingestionProperties.setReportMethod(QueueAndTable);
+                ingestionProperties.setReportLevel(IngestionProperties.IngestionReportLevel.FailuresAndSuccesses);
+                FileSourceInfo fileSourceInfo = new FileSourceInfo(System.getProperty("filePath"), 0);
+                ingestionResult = client.ingestFromFile(fileSourceInfo, ingestionProperties);
+            }
```

Comment:
Catch?

Reply:
In Java, a try-with-resources block is effectively a cleaner and more concise way to include a finally block. The catch is optional, and superfluous if unnecessary: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html